### PR TITLE
Resources PR

### PR DESCRIPTION
**Pull request by ishranga11**
- Removed Hashcode and equals function from MultipleChoiceQuestion to use parent functions.
- Updated ExamInstance equals function to allow not null answers comparison
- Added Answer model constructor for tests
- Added Exceptions, requests, responses classes in resources/io
- Added Tests for resources
- Update pom.xml to add dropwizard testing
- Updated schema.sql to allow all dao tests to work simultaneously.
- Removed duplicating license in schema.sql

### CODE REVIEW

File: appengine/src/main/java/com/google/googleinterns/gscribe/models/ExamInstance.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public boolean equals(Object o) {
                 getUserID().equals(that.getUserID()) &&
                 Objects.equals(getStartTime(), that.getStartTime()) &&
                 Objects.equals(getEndTime(), that.getEndTime()) &&
-                getAnswers().equals(that.getAnswers());
+                Objects.equals(getAnswers(), that.getAnswers());
```
Comment (dhruvpancholi) [inline appengine/src/main/java/com/google/googleinterns/gscribe/models/ExamInstance.java]: Don't use a getter within the same class.

Comment (ishranga11) [reply appengine/src/main/java/com/google/googleinterns/gscribe/models/ExamInstance.java]: Updated.

File: appengine/src/main/java/com/google/googleinterns/gscribe/models/MultipleChoiceQuestion.java [hunk 2]
```diff
@@ -39,16 +38,4 @@ public void setOptions(List<String> options) {
         this.options = options;
     }
 
-    @Override
```
Comment (dhruvpancholi) [inline appengine/src/main/java/com/google/googleinterns/gscribe/models/MultipleChoiceQuestion.java]: If you remove this, it won't be possible to check if two MultipleChoiceQuestions are same along with options. The correct way of doing it with inheritance can be found here: https://stackoverflow.com/questions/2066917/overriding-equals-hashcode-in-sub-classes-considering-super-fields

File: appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java [hunk 3]
```diff
@@ -16,61 +16,262 @@
...
      */
     @POST
-    public ExamResponse postExam(@HeaderParam("authorization-code") String IDToken, ExamRequest request) {
-        return null;
+    public ExamResponse postExam(@NotNull @HeaderParam("Authentication") String IDToken, @NotNull ExamRequest request) {
+        String userID;
+        User token;
+        Exam exam;
+        try {
+            userID = tokenService.verifyIDToken(IDToken);
```
Comment (dhruvpancholi) [inline appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java:102]: A lot of duplicate code, can you please all the try catch to the service itself. This method will be throwing all the server related exceptions.

File: appengine/src/main/java/com/google/googleinterns/gscribe/models/MultipleChoiceQuestion.java (hunk 2 above)
Comment (ishranga11) [reply appengine/src/main/java/com/google/googleinterns/gscribe/models/MultipleChoiceQuestion.java]: Added the equals and hashcode function.

File: appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java (hunk 3 above)
Comment (ishranga11) [reply appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java:102]: Added a TODO for this.

File: appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java [hunk 4]
```diff
@@ -16,61 +16,263 @@
...
+        this.tokenService = tokenService;
+        this.userTokenDao = userTokenDao;
+        this.examMetadataDao = examMetadataDao;
+        this.questionsDao = questionsDao;
+        this.examInstanceDao = examInstanceDao;
+        this.answerDao = answerDao;
+        this.spreadsheetService = spreadsheetService;
+        this.objectMapper = objectMapper;
+    }
+    // TODO: Code cleanup required
```
Comment (dhruvpancholi) [inline appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java]: Be more specific, what all exact changes will be required.

**Review by dhruvpancholi — CHANGES_REQUESTED** [review]

File: appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java (hunk 4 above)
Comment (ishranga11) [reply appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java]: Updated.

**Review by dhruvpancholi — APPROVED** [review]
There are still a few improvements in this PR, but considering the fact that Friday is the last day for internship, merging it to be code complete.

**Pull request merged**